app/lib/features/chat/pages/room_profile_page.dart

```diff
@@ -139,9 +139,10 @@ class RoomProfilePage extends ConsumerWidget {
                   tiles: [
                     SettingsTile(
                       onPressed: (ctx) {
+                        final serverName = roomId.split(':').last;
```

Comment:
This isn't really how `via` is meant to work. See the [matrix URI spec](https://spec.matrix.org/v1.9/appendices/#uris) for details. But ideally `via` tells you about (additional) servers that we know to be in the room as alternative ways to connect to the room (in particular if the homeserver isn't available). Just repeating the homeserver is a bit of a short-cut that doesn't in practice actually help doing the right resolution.

We can keep this for now, but should probably add a helper function on the rust side that compiles the actual correct information including `via`.

Reply:
Ok got it. I will check for matrix specification and do required changes accordingly.